### ISSUE TITLE
thrift: TZlibTransport: possible memory leak when running ThriftTest

### ISSUE THREAD

**Issue by cfriedt** [cfriedt/thrift-for-zephyr#160]
**Describe the bug**
It looks like there may be a memory leak in TZlibTransport. My guess is that this is quite specific to the Zephyr implementation, similarly to the file descriptor leak previously reported (which was due to some C++ issue). I added a simple `mallinfo()` implementation that allowed me to instrument heap usage. The newlib `mallinfo()` does not work unfortunately in Zephyr (it seems to report garbage). 

Please also mention any information which could help others to understand
the problem you're facing:
 - **What target platform are you using?** `qemu_x86_64`
 - **What have you tried to diagnose or workaround this issue?** See description

**To Reproduce**
1. Apply [mallinfo.patch](https://github.com/zephyrproject-rtos/gsoc-2022-thrift/files/10466626/mallinfo.patch)
2. `west build -p auto -b qemu_x86_64 -t debugserver ../modules/lib/thrift/tests/lib/thrift/ThriftTest/`
3. See error

**Expected behavior**
No memory leak

**Impact**
It's a significant bug that should be fixed before the module is merged upstream to Zephyr

**Logs and console output**
```
*** Booting Zephyr OS build zephyr-v3.2.0-3493-g759dc8d3bcb4 ***
Running TESTSUITE thrift
===================================================================
START - test_binary
Memory stats Before: {arena: 1503232, uordblks: 12288, fordblks: 1490944, usage: 0%}
deflateInit: 0x68f540 -1
deflateInit: 0x68f4c0 -1
MALLOC: 48@0x6a9de0: deflateInit(): 134
MALLOC: 48@0x6a9e20: deflateInit(): 134
MALLOC: 32768@0x6a9e60: deflateInit(): 145
MALLOC: 32768@0x6b1e70: deflateInit(): 145
deflateInit: 0x6ba920 -1
MALLOC: 48@0x6c7db0: deflateInit(): 134
MALLOC: 32768@0x6c7df0: deflateInit(): 145
REALLOC: 64@0x6d08c0: outbits(): 113
FREE: 0x6d08c0: deflate(): 275
testBinary(0: )
REALLOC: 64@0x6d0dd0: outbits(): 113
FREE: 0x6d0dd0: deflate(): 275
REALLOC: 64@0x6d0e40: outbits(): 113
FREE: 0x6d0e40: deflate(): 275
testBinary(5: Hello)
REALLOC: 64@0x6d0e90: outbits(): 113
FREE: 0x6d0e90: deflate(): 275
REALLOC: 64@0x6d0ee0: outbits(): 113
FREE: 0x6d0ee0: deflate(): 275
testBinary(4: H)
REALLOC: 64@0x6d0f30: outbits(): 113
FREE: 0x6d0f30: deflate(): 275
caught transport exception
server was interrupted
FREE: 0x6b1e70: deflateEnd(): 281
FREE: 0x6a9e20: deflateEnd(): 282
Memory stats After: {arena: 1503232, uordblks: 376832, fordblks: 1126400, usage: 25%}
 PASS - test_binary in 0.056 seconds
===================================================================
START - test_bool
Memory stats Before: {arena: 1503232, uordblks: 376832, fordblks: 1126400, usage: 25%}
deflateInit: 0x68ff30 -1
MALLOC: 48@0x68e060: deflateInit(): 134
MALLOC: 32768@0x6e8630: deflateInit(): 145
deflateInit: 0x69fb10 -1
MALLOC: 48@0x6d0550: deflateInit(): 134
MALLOC: 32768@0x6f8650: deflateInit(): 145
deflateInit: 0x6a06d0 -1
MALLOC: 48@0x6a0750: deflateInit(): 134
MALLOC: 32768@0x70da70: deflateInit(): 145
REALLOC: 64@0x6a0b90: outbits(): 113
FREE: 0x6a0b90: deflate(): 275
testBool(false)
REALLOC: 64@0x6a0f30: outbits(): 113
FREE: 0x6a0f30: deflate(): 275
REALLOC: 64@0x6a0fa0: outbits(): 113
FREE: 0x6a0fa0: deflate(): 275
testBool(true)
REALLOC: 64@0x6a0ff0: outbits(): 113
FREE: 0x6a0ff0: deflate(): 275
caught transport exception
server was interrupted
FREE: 0x6e8630: deflateEnd(): 281
FREE: 0x68e060: deflateEnd(): 282
Memory stats After: {arena: 1503232, uordblks: 585728, fordblks: 917504, usage: 38%}
 PASS - test_bool in 0.009 seconds
===================================================================
START - test_byte
Memory stats Before: {arena: 1503232, uordblks: 585728, fordblks: 917504, usage: 38%}
deflateInit: 0x68e400 -1
MALLOC: 48@0x6d0880: deflateInit(): 134
MALLOC: 32768@0x722e90: deflateInit(): 145
deflateInit: 0x690860 -1
MALLOC: 48@0x6a1a60: deflateInit(): 134
MALLOC: 32768@0x72aea0: deflateInit(): 145
deflateInit: 0x691220 -1
MALLOC: 48@0x6912a0: deflateInit(): 134
MALLOC: 32768@0x7402c0: deflateInit(): 145
REALLOC: 64@0x691a10: outbits(): 113
FREE: 0x691a10: deflate(): 275
testByte(0)
REALLOC: 64@0x691a80: outbits(): 113
FREE: 0x691a80: deflate(): 275
REALLOC: 64@0x691af0: outbits(): 113
FREE: 0x691af0: deflate(): 275
testByte(-1)
REALLOC: 64@0x691b40: outbits(): 113
FREE: 0x691b40: deflate(): 275
REALLOC: 64@0x691b90: outbits(): 113
FREE: 0x691b90: deflate(): 275
testByte(42)
REALLOC: 64@0x691be0: outbits(): 113
FREE: 0x691be0: deflate(): 275
REALLOC: 64@0x691c30: outbits(): 113
FREE: 0x691c30: deflate(): 275
testByte(-42)
REALLOC: 64@0x691c80: outbits(): 113
FREE: 0x691c80: deflate(): 275
REALLOC: 64@0x691cd0: outbits(): 113
FREE: 0x691cd0: deflate(): 275
testByte(127)
REALLOC: 64@0x691d20: outbits(): 113
FREE: 0x691d20: deflate(): 275
REALLOC: 64@0x691d70: outbits(): 113
FREE: 0x691d70: deflate(): 275
testByte(-128)
REALLOC: 64@0x691dc0: outbits(): 113
FREE: 0x691dc0: deflate(): 275
caught transport exception
server was interrupted
FREE: 0x722e90: deflateEnd(): 281
FREE: 0x6d0880: deflateEnd(): 282
Memory stats After: {arena: 1503232, uordblks: 794624, fordblks: 708608, usage: 52%}
 PASS - test_byte in 0.012 seconds
===================================================================
START - test_double
Memory stats Before: {arena: 1503232, uordblks: 794624, fordblks: 708608, usage: 52%}
deflateInit: 0x692b60 -1
MALLOC: 48@0x692710: deflateInit(): 134
MALLOC: 32768@0x7482d0: deflateInit(): 145
deflateInit: 0x6936b0 -1
MALLOC: 48@0x6a1960: deflateInit(): 134
MALLOC: 32768@0x75d6f0: deflateInit(): 145
deflateInit: 0x693f50 -1
REALLOC: 64@0x68e380: outbits(): 113
MALLOC: 48@0x68e3d0: deflateInit(): 134
MALLOC: 32768@0x772b10: deflateInit(): 145
FREE: 0x68e380: deflate(): 275
testDouble(0.000000)
REALLOC: 64@0x694720: outbits(): 113
FREE: 0x694720: deflate(): 275
REALLOC: 64@0x694790: outbits(): 113
FREE: 0x694790: deflate(): 275
testDouble(-1.000000)
REALLOC: 64@0x68e470: outbits(): 113
FREE: 0x68e470: deflate(): 275
REALLOC: 64@0x6947e0: outbits(): 113
FREE: 0x6947e0: deflate(): 275
testDouble(-5.209852)
REALLOC: 64@0x694830: outbits(): 113
FREE: 0x694830: deflate(): 275
REALLOC: 64@0x694880: outbits(): 113
FREE: 0x694880: deflate(): 275
testDouble(-0.000341)
REALLOC: 64@0x6948d0: outbits(): 113
FREE: 0x6948d0: deflate(): 275
REALLOC: 64@0x694920: outbits(): 113
FREE: 0x694920: deflate(): 275
testDouble(179769313486231570814527423731704356798070567525844996598917476803157260780028538760589558632766878171540458953514382464234321326889464182768467546703537516986049910576551282076245490090389328944075868508455133942304583236903222948165808559332123348274797826204144723168738177180919299881250404026184124858368.000000)
REALLOC: 64@0x77b440: outbits(): 113
FREE: 0x77b440: deflate(): 275
REALLOC: 64@0x77b490: outbits(): 113
FREE: 0x77b490: deflate(): 275
testDouble(-179769313486231570814527423731704356798070567525844996598917476803157260780028538760589558632766878171540458953514382464234321326889464182768467546703537516986049910576551282076245490090389328944075868508455133942304583236903222948165808559332123348274797826204144723168738177180919299881250404026184124858368.000000)
REALLOC: 64@0x77b510: outbits(): 113
FREE: 0x77b510: deflate(): 275
caught transport exception
server was interrupted
FREE: 0x7482d0: deflateEnd(): 281
FREE: 0x692710: deflateEnd(): 282
Memory stats After: {arena: 1503232, uordblks: 1003520, fordblks: 499712, usage: 66%}
 PASS - test_double in 0.020 seconds
===================================================================
START - test_enum
Memory stats Before: {arena: 1503232, uordblks: 1003520, fordblks: 499712, usage: 66%}
deflateInit: 0x69a730 -1
MALLOC: 48@0x6a0ed0: deflateInit(): 134
deflateInit: 0x6e0ec0 -1
MALLOC: 32768@0x790980: deflateInit(): 145
MALLOC: 48@0x6929f0: deflateInit(): 134
MALLOC: 32768@0x798990: deflateInit(): 145
deflateInit: 0x6e1bf0 -1
MALLOC: 48@0x68f960: deflateInit(): 134
MALLOC: 32768@0x7addb0: deflateInit(): 145
REALLOC: 64@0x6e1c70: outbits(): 113
FREE: 0x6e1c70: deflate(): 275
testEnum(1)
REALLOC: 64@0x6e2410: outbits(): 113
FREE: 0x6e2410: deflate(): 275
REALLOC: 64@0x6e2480: outbits(): 113
FREE: 0x6e2480: deflate(): 275
testEnum(2)
REALLOC: 64@0x6e24d0: outbits(): 113
FREE: 0x6e24d0: deflate(): 275
REALLOC: 64@0x6e2520: outbits(): 113
FREE: 0x6e2520: deflate(): 275
testEnum(8)
REALLOC: 64@0x6e2570: outbits(): 113
FREE: 0x6e2570: deflate(): 275
caught transport exception
server was interrupted
FREE: 0x790980: deflateEnd(): 281
FREE: 0x6a0ed0: deflateEnd(): 282
Memory stats After: {arena: 1503232, uordblks: 1249280, fordblks: 253952, usage: 83%}
 PASS - test_enum in 0.012 seconds
===================================================================
START - test_exception
Memory stats Before: {arena: 1503232, uordblks: 1249280, fordblks: 253952, usage: 83%}
deflateInit: 0x68fa70 -1
MALLOC: 48@0x6919d0: deflateInit(): 134
MALLOC: 32768@0x7b5dc0: deflateInit(): 145
deflateInit: 0x69b170 -1
MALLOC: 48@0x6e2fe0: deflateInit(): 134
MALLOC: 32768@0x7cb1e0: deflateInit(): 145
deflateInit: 0x69bc90 -1
MALLOC: 48@0x6e1280: deflateInit(): 134
MALLOC: 32768@0x7e0600: deflateInit(): 145
REALLOC: 64@0x69c440: outbits(): 113
FREE: 0x69c440: deflate(): 275
testException(Xception)
REALLOC: 64@0x69c4b0: outbits(): 113
FREE: 0x69c4b0: deflate(): 275
REALLOC: 64@0x69c520: outbits(): 113
FREE: 0x69c520: deflate(): 275
testException(TException)
REALLOC: 64@0x69c660: outbits(): 113
FREE: 0x69c660: deflate(): 275
REALLOC: 64@0x69c790: outbits(): 113
FREE: 0x69c790: deflate(): 275
testException(success)
REALLOC: 64@0x69c7e0: outbits(): 113
FREE: 0x69c7e0: deflate(): 275
caught transport exception
server was interrupted
FREE: 0x7b5dc0: deflateEnd(): 281
FREE: 0x6919d0: deflateEnd(): 282
Memory stats After: {arena: 1503232, uordblks: 1458176, fordblks: 45056, usage: 97%}
 PASS - test_exception in 0.012 seconds
===================================================================
START - test_i32
Memory stats Before: {arena: 1503232, uordblks: 1458176, fordblks: 45056, usage: 97%}
deflateInit: 0x69d4c0 -1
MALLOC: 48@0x68fd90: deflateInit(): 134
deflateInit: 0x69de70 -1
MALLOC: 48@0x69d0d0: deflateInit(): 134
caught transport exception
exit
Thrift: Thu Jan  1 00:00:00 1970 TServerTransport died: TServerTransport died: 
qemu-system-x86_64: terminating on signal 2
ninja: build stopped: interrupted by user.
```

Mallinfo change is attached. Aside from that, I just added printk logging in a couple of files.

**Environment (please complete the following information):**
 - OS: CI (Linux)
 - Toolchain: Zephyr SDK 0.15.2
 - Commit SHA or Version used: 17a695c86e060c99872a1b8e000e066490bd793a

**Additional context**
This was discovered in the context of switching to upstream Apache Thrift (after many required POSIX changes were merged into Zephyr)

**Comment by cfriedt** [comment]
There is a workaround for this in #158

**Comment by cfriedt** [comment]
After Fix:
```
*** Booting Zephyr OS build zephyr-v3.2.0-3493-g759dc8d3bcb4 ***
Running TESTSUITE thrift
===================================================================
START - test_binary
Memory stats Before: {arena: 1589248, uordblks: 12288, fordblks: 1576960, usage: 0%}
server thread is 0x23f120
testBinary(0: )
testBinary(5: Hello)
testBinary(4: H)
calling inflateEnd()
calling inflateEnd()
calling reset() on shared_ptrs
calling inflateEnd()
finished calling reset() on shared_ptrs
Memory stats After: {arena: 1589248, uordblks: 376832, fordblks: 1212416, usage: 23%}
 PASS - test_binary in 0.053 seconds
===================================================================
START - test_bool
Memory stats Before: {arena: 1589248, uordblks: 376832, fordblks: 1212416, usage: 23%}
server thread is 0x23f120
testBool(false)
testBool(true)
calling inflateEnd()
calling inflateEnd()
calling reset() on shared_ptrs
calling inflateEnd()
finished calling reset() on shared_ptrs
Memory stats After: {arena: 1589248, uordblks: 376832, fordblks: 1212416, usage: 23%}
 PASS - test_bool in 0.009 seconds
===================================================================
START - test_byte
Memory stats Before: {arena: 1589248, uordblks: 376832, fordblks: 1212416, usage: 23%}
server thread is 0x23f120
testByte(0)
testByte(-1)
testByte(42)
testByte(-42)
testByte(127)
testByte(-128)
calling inflateEnd()
calling inflateEnd()
calling reset() on shared_ptrs
calling inflateEnd()
finished calling reset() on shared_ptrs
Memory stats After: {arena: 1589248, uordblks: 376832, fordblks: 1212416, usage: 23%}
 PASS - test_byte in 0.008 seconds
===================================================================
START - test_double
Memory stats Before: {arena: 1589248, uordblks: 376832, fordblks: 1212416, usage: 23%}
server thread is 0x23f120
testDouble(0.000000)
testDouble(-1.000000)
testDouble(-5.209852)
testDouble(-0.000341)
testDouble(179769313486231570814527423731704356798070567525844996598917476803157260780028538760589558632766878171540458953514382464234321326889464182768467546703537516986049910576551282076245490090389328944075868508455133942304583236903222948165808559332123348274797826204144723168738177180919299881250404026184124858368.000000)
testDouble(-179769313486231570814527423731704356798070567525844996598917476803157260780028538760589558632766878171540458953514382464234321326889464182768467546703537516986049910576551282076245490090389328944075868508455133942304583236903222948165808559332123348274797826204144723168738177180919299881250404026184124858368.000000)
calling inflateEnd()
calling inflateEnd()
calling reset() on shared_ptrs
calling inflateEnd()
finished calling reset() on shared_ptrs
Memory stats After: {arena: 1589248, uordblks: 376832, fordblks: 1212416, usage: 23%}
 PASS - test_double in 0.015 seconds
===================================================================
START - test_enum
Memory stats Before: {arena: 1589248, uordblks: 376832, fordblks: 1212416, usage: 23%}
server thread is 0x23f120
testEnum(1)
testEnum(2)
testEnum(8)
calling inflateEnd()
calling inflateEnd()
calling reset() on shared_ptrs
calling inflateEnd()
finished calling reset() on shared_ptrs
Memory stats After: {arena: 1589248, uordblks: 376832, fordblks: 1212416, usage: 23%}
 PASS - test_enum in 0.006 seconds
===================================================================
START - test_exception
Memory stats Before: {arena: 1589248, uordblks: 376832, fordblks: 1212416, usage: 23%}
server thread is 0x23f120
testException(Xception)
testException(TException)
testException(success)
calling inflateEnd()
calling inflateEnd()
calling reset() on shared_ptrs
calling inflateEnd()
finished calling reset() on shared_ptrs
Memory stats After: {arena: 1589248, uordblks: 376832, fordblks: 1212416, usage: 23%}
 PASS - test_exception in 0.008 seconds
===================================================================
START - test_i32
Memory stats Before: {arena: 1589248, uordblks: 376832, fordblks: 1212416, usage: 23%}
server thread is 0x23f120
testI32(0)
testI32(-1)
testI32(190000013)
testI32(-190000013)
testI32(2147483647)
testI32(-2147483648)
calling inflateEnd()
calling inflateEnd()
calling reset() on shared_ptrs
calling inflateEnd()
finished calling reset() on shared_ptrs
Memory stats After: {arena: 1589248, uordblks: 376832, fordblks: 1212416, usage: 23%}
 PASS - test_i32 in 0.009 seconds
===================================================================
START - test_i64
Memory stats Before: {arena: 1589248, uordblks: 376832, fordblks: 1212416, usage: 23%}
server thread is 0x23f120
testI64(0)
testI64(-1)
testI64(7000000000000000123)
testI64(-7000000000000000123)
testI64(9223372036854775807)
testI64(-9223372036854775808)
calling inflateEnd()
calling inflateEnd()
calling reset() on shared_ptrs
calling inflateEnd()
finished calling reset() on shared_ptrs
Memory stats After: {arena: 1589248, uordblks: 376832, fordblks: 1212416, usage: 23%}
 PASS - test_i64 in 0.009 seconds
===================================================================
START - test_list
Memory stats Before: {arena: 1589248, uordblks: 376832, fordblks: 1212416, usage: 23%}
server thread is 0x23f120
testList({})
testList({-2, -1, 0, 1, 2})
calling inflateEnd()
calling inflateEnd()
calling reset() on shared_ptrs
calling inflateEnd()
finished calling reset() on shared_ptrs
Memory stats After: {arena: 1589248, uordblks: 376832, fordblks: 1212416, usage: 23%}
 PASS - test_list in 0.009 seconds
===================================================================
START - test_map
Memory stats Before: {arena: 1589248, uordblks: 376832, fordblks: 1212416, usage: 23%}
server thread is 0x23f120
testMap({0 => -10, 1 => -9, 2 => -8, 3 => -7, 4 => -6})
calling inflateEnd()
calling inflateEnd()
calling reset() on shared_ptrs
calling inflateEnd()
finished calling reset() on shared_ptrs
Memory stats After: {arena: 1589248, uordblks: 376832, fordblks: 1212416, usage: 23%}
 PASS - test_map in 0.009 seconds
===================================================================
START - test_multi_exception
Memory stats Before: {arena: 1589248, uordblks: 376832, fordblks: 1212416, usage: 23%}
server thread is 0x23f120
testMultiException(Xception, test 1)
testMultiException(Xception2, test 2)
calling inflateEnd()
calling inflateEnd()
calling reset() on shared_ptrs
calling inflateEnd()
finished calling reset() on shared_ptrs
Memory stats After: {arena: 1589248, uordblks: 376832, fordblks: 1212416, usage: 23%}
 PASS - test_multi_exception in 0.007 seconds
===================================================================
START - test_nested_map
Memory stats Before: {arena: 1589248, uordblks: 376832, fordblks: 1212416, usage: 23%}
server thread is 0x23f120
testMapMap(1)
calling inflateEnd()
calling inflateEnd()
calling reset() on shared_ptrs
calling inflateEnd()
finished calling reset() on shared_ptrs
Memory stats After: {arena: 1589248, uordblks: 376832, fordblks: 1212416, usage: 23%}
 PASS - test_nested_map in 0.009 seconds
===================================================================
START - test_nested_struct
Memory stats Before: {arena: 1589248, uordblks: 376832, fordblks: 1212416, usage: 23%}
server thread is 0x23f120
testNest({1, {"Zero", 1, -3, -5}, 5})
calling inflateEnd()
calling inflateEnd()
calling reset() on shared_ptrs
calling inflateEnd()
finished calling reset() on shared_ptrs
Memory stats After: {arena: 1589248, uordblks: 376832, fordblks: 1212416, usage: 23%}
 PASS - test_nested_struct in 0.006 seconds
===================================================================
START - test_set
Memory stats Before: {arena: 1589248, uordblks: 376832, fordblks: 1212416, usage: 23%}
server thread is 0x23f120
testSet({-2, -1, 0, 1, 2})
calling inflateEnd()
calling inflateEnd()
calling reset() on shared_ptrs
calling inflateEnd()
finished calling reset() on shared_ptrs
Memory stats After: {arena: 1589248, uordblks: 376832, fordblks: 1212416, usage: 23%}
 PASS - test_set in 0.008 seconds
===================================================================
START - test_string
Memory stats Before: {arena: 1589248, uordblks: 376832, fordblks: 1212416, usage: 23%}
server thread is 0x23f120
testString("Test")
calling inflateEnd()
calling inflateEnd()
calling reset() on shared_ptrs
calling inflateEnd()
finished calling reset() on shared_ptrs
Memory stats After: {arena: 1589248, uordblks: 376832, fordblks: 1212416, usage: 23%}
 PASS - test_string in 0.006 seconds
===================================================================
START - test_string_map
Memory stats Before: {arena: 1589248, uordblks: 376832, fordblks: 1212416, usage: 23%}
server thread is 0x23f120
testMap({a => 2, b => blah, some => thing})
calling inflateEnd()
calling inflateEnd()
calling reset() on shared_ptrs
calling inflateEnd()
finished calling reset() on shared_ptrs
Memory stats After: {arena: 1589248, uordblks: 376832, fordblks: 1212416, usage: 23%}
 PASS - test_string_map in 0.008 seconds
===================================================================
START - test_struct
Memory stats Before: {arena: 1589248, uordblks: 376832, fordblks: 1212416, usage: 23%}
server thread is 0x23f120
testStruct({"Zero", 1, -3, -5})
calling inflateEnd()
calling inflateEnd()
calling reset() on shared_ptrs
calling inflateEnd()
finished calling reset() on shared_ptrs
Memory stats After: {arena: 1589248, uordblks: 376832, fordblks: 1212416, usage: 23%}
 PASS - test_struct in 0.007 seconds
===================================================================
START - test_typedef
Memory stats Before: {arena: 1589248, uordblks: 376832, fordblks: 1212416, usage: 23%}
server thread is 0x23f120
testTypedef(309858235082523)
calling inflateEnd()
calling inflateEnd()
calling reset() on shared_ptrs
calling inflateEnd()
finished calling reset() on shared_ptrs
Memory stats After: {arena: 1589248, uordblks: 376832, fordblks: 1212416, usage: 23%}
 PASS - test_typedef in 0.005 seconds
===================================================================
START - test_void
Memory stats Before: {arena: 1589248, uordblks: 376832, fordblks: 1212416, usage: 23%}
server thread is 0x23f120
testVoid()
calling inflateEnd()
calling inflateEnd()
calling reset() on shared_ptrs
calling inflateEnd()
finished calling reset() on shared_ptrs
Memory stats After: {arena: 1589248, uordblks: 376832, fordblks: 1212416, usage: 23%}
 PASS - test_void in 0.005 seconds
===================================================================
TESTSUITE thrift succeeded

------ TESTSUITE SUMMARY START ------

SUITE PASS - 100.00% [thrift]: pass = 19, fail = 0, skip = 0, total = 19 duration = 0.196 seconds
 - PASS - [thrift.test_binary] duration = 0.053 seconds
 - PASS - [thrift.test_bool] duration = 0.009 seconds
 - PASS - [thrift.test_byte] duration = 0.008 seconds
 - PASS - [thrift.test_double] duration = 0.015 seconds
 - PASS - [thrift.test_enum] duration = 0.006 seconds
 - PASS - [thrift.test_exception] duration = 0.008 seconds
 - PASS - [thrift.test_i32] duration = 0.009 seconds
 - PASS - [thrift.test_i64] duration = 0.009 seconds
 - PASS - [thrift.test_list] duration = 0.009 seconds
 - PASS - [thrift.test_map] duration = 0.009 seconds
 - PASS - [thrift.test_multi_exception] duration = 0.007 seconds
 - PASS - [thrift.test_nested_map] duration = 0.009 seconds
 - PASS - [thrift.test_nested_struct] duration = 0.006 seconds
 - PASS - [thrift.test_set] duration = 0.008 seconds
 - PASS - [thrift.test_string] duration = 0.006 seconds
 - PASS - [thrift.test_string_map] duration = 0.008 seconds
 - PASS - [thrift.test_struct] duration = 0.007 seconds
 - PASS - [thrift.test_typedef] duration = 0.005 seconds
 - PASS - [thrift.test_void] duration = 0.005 seconds

------ TESTSUITE SUMMARY END ------

===================================================================
PROJECT EXECUTION SUCCESSFUL
```

**Comment by cfriedt** [comment]
A workaround is included as part of #158
Upstream issue is
https://issues.apache.org/jira/browse/THRIFT-5678
